x-pack/plugins/monitoring/server/types.ts

```diff
@@ -257,3 +250,5 @@ export interface PipelineVersion {
   lastSeen: number;
   hash: string;
 }
+
+export type MonitoringConfigSchema = TypeOf<typeof configSchema>;
```

Comment:
I had a hard time deciding what to call this. If I follow core it would be called `MonitoringConfigType` (https://github.com/elastic/kibana/blob/1eea7a32d4cac9a125c76eb967f05bd9d034ed2d/src/core/server/elasticsearch/elasticsearch_config.ts#L22) but that seemed like a mismatch with the typeof input (`configSchema`).

As far as I can see, we need to isolate `MonitoringConfigSchema` (the input config) from `MonitoringConfig` (the config used across the app) due to the creation of a `MonitoringElasticsearchConfig` object at `ui.elasticsearch` in the structure.

Reply:
Perhaps this is less critical since the code mainly refers to `MonitoringConfig`?

Reply:
👍🏻  I like that logic